### PR TITLE
Bonus activity spread operator and optional chaining 

### PR DESCRIPTION
**Pull request by EllieJudge**
- adds bonus docs, bonus katas and bonus tests on the spread operator and optional chaining

### CODE REVIEW

File: docs/exercise007.md [hunk 1]
```diff
@@ -14,4 +14,6 @@ In these exercises you have covered forLoops, ES6 methods like .map() and .forEa
 
 Great work, we hope you found these exercises fun and look forward to seeing you soon. 🙌
 
+See Exercise008 for an optional Bonus activity!
```
Comment (eggsy84) [inline docs/exercise007.md]: For ease, you could make this a link to the Exercise008 document

File: docs/exercise008-bonus-activity.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+Congratulations for finishing these JavaScript challenges!
+
+You have covered so many different JavaScript concepts and skills in a very short space of time, including:
+- Problem Solving
+- For Loops
+- ES6 methods
+- Conditionals
+- Data Types
+- Objects
+and more!
```
Comment (eggsy84) [inline docs/exercise008-bonus-activity.md:12]: I like the re-enforcement here covering what they have learnt to say well done to them as well 👍

File: docs/exercise008-bonus-activity.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+In these exercise we would like you to take a look at two new features of ES6, the Spread Operator and Optional Chaining.
```
Comment (eggsy84) [inline docs/exercise008-bonus-activity.md]: Think there is a typo here "exercises"

The optional chaining was introduced in ES2020 I think @helloheatherw can confirm maybe?

So we might want to say try some of the very latest JavaScript and have a go with Optional chaining

File: docs/exercise008-bonus-activity.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+catUser.miceEaten?.[]; // 5
+
+dogUser.miceEaten?.[]; // undefined (dogs don't eat mice, silly! 🐭)
+```
+
+
+1.1) Work through Exercises 008 using ✨Optional chaining✨
+
+When it comes to working with React you will find spread syntax, optional chaining and methods like .map() and .filter() help a lot. We hope you enjoyed these exercises. Happy coding!
```
Comment (eggsy84) [inline docs/exercise008-bonus-activity.md]: I think (at time of writing) optional chaining wont be fully supported out of the box with React apps created using "create-react-app" - maybe @helloheatherw could validate that?

On the assumption above I would maybe put some context about the react statement and that optional chaining is not yet fully supported.

"Later on in the programme you'll find we work through React and you will find spread syntax methods like .map() and .filter() help a lot. Note: Optional chaining is very new as part of [ES2020](https://www.freecodecamp.org/news/javascript-new-features-es2020/) so is not yet in the mainstream React. We hope you enjoyed these exercises. Happy coding!"

File: docs/exercise008-bonus-activity.md (hunk 3 above)
Comment (EllieJudge) [reply docs/exercise008-bonus-activity.md]: Ahhh brill, I meant to ask/confirm this and completely forgot! I'd googled it but couldn't find when exactly it came out. I'll update :)

File: docs/exercise008-bonus-activity.md (hunk 4 above)
Comment (eggsy84) [reply docs/exercise008-bonus-activity.md]: Hey @EllieJudge 

Totally ignore this comment above. It looks like the latest versions do support it you're right 😍

https://adostes.medium.com/enabling-optional-chaining-in-a-create-react-app-a9f626a515d9

The pointers at the top show that create-react-app v3.3.0 supports optional chaining whoop whoop!

File: test/exercise008-bonus-activity.test.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+describe("addressLookUp", () => {
+    test("returns a users postcode", () => {
+
+        const user = {
+            name: 'JimBob Eggs',
+            telephone: 666,
+            likes: 'cheese',
+            dislikes: 'dirty socks',
+            address: {
+                houseNumber: 'The Number of the Beast',
```
Comment (eggsy84) [inline test/exercise008-bonus-activity.test.js:61]: #notoffendedatall 😳

**Pull request merged**